### ISSUE TITLE
QUESTION: What permissions are required to see FK links in a table's data view?

### ISSUE THREAD

**Issue by scottdillon** [beekeeper-studio/beekeeper-studio#436]
**Details:**
I have set up a read-only user for engineers to access several Postgres databases my company uses. The read-only user does not see graphic key indicators that a column is a primary or foreign key. The read-only user also does not have FK links to the FK table in the FK column's rows.

My superuser login, however **does** see both key icons at the top of a PK and FK column and FK link buttons in the rows.

What are the permissions I need to set for engineer users to be able to see these details. I do not want to give them update, delete or insert permissions for tables.

Thanks for your work. Beekeeper studio is great. Look forward to see where it goes.

**Version Information [might be relevant to your issue]**
Operating System: MacOS 10.15.7
App Version [help -> about]: 1.8.7

**Screenshot / GIF [if appropriate]**:
Read-only user:
![Screen Shot 2020-11-09 at 4 48 29 PM](https://user-images.githubusercontent.com/6405083/98600807-237c4c80-22ac-11eb-9a0e-0aefd6bf9782.png)

Superadmin:
![Screen Shot 2020-11-09 at 4 51 11 PM](https://user-images.githubusercontent.com/6405083/98600844-31ca6880-22ac-11eb-8e44-7f90cf5a4067.png)

**Comment by marko-ilic** [comment]
Hello,
I have the same issue. The user with read-only permissions cannot see relations. 
Works fine for the user with the "full access".

DB: Postgres
Beekeeper Studio 3.2.1
Mac OS 12.2.1

**Comment by rathboma** [comment]
Hey! Yes this is an issue. I'll be addressing it soon. Sorry for this mistake, I wasn't aware that the views I was using had such limitations.

**Comment by wmontgomery** [comment]
Picking this up and going to start looking into it. Stay tuned!

**Comment by rathboma** [comment]
Great!

**Comment by rathboma** [comment]
@marko-ilic does this work in other tools like DBeaver?

**Comment by marko-ilic** [comment]
> @marko-ilic does this work in other tools like DBeaver?

Yes, it works in DBeaver https://github.com/beekeeper-studio/beekeeper-studio/issues/1140 (the same issue)

**Comment by marko-ilic** [comment]
> @marko-ilic Do you know what query I need to use to get the relations for the read-only use? Having trouble fixing this.

Maybe this one
```sql
SELECT conname, pg_catalog.pg_get_constraintdef(r.oid, true) as condef
FROM pg_catalog.pg_constraint r
WHERE r.conrelid = 'tableName'::regclass
```
